### ISSUE TITLE
false positives from delete

### ISSUE THREAD

**Issue by wojdyr** [xiw/stack#14]
I get a lot of false positives from `delete`. Apparently the `icmp eq` below comes from `delete`. Can I do anything about it?

```
[wojdyr@dt mw]$ cat delete-test.cc 
#include <stdio.h>

struct Ab { int a; };

int main()
{
    Ab *x = new Ab;
    x->a = 123;
    printf("%d\n", x->a);
    delete x;
    return 0;
}
[wojdyr@dt mw]$ stack-build clang++ delete-test.cc
[wojdyr@dt mw]$ poptck 
Analyzing ./delete-test.17250.ll
Generated 1 warnings, see pstack.txt for details.
[wojdyr@dt mw]$ cat pstack.txt

---
bug: anti-simplify
model: |
  %7 = icmp eq %struct.Ab* %3, null, !dbg !19
  -->  false
stack: 
  - /run/mw/delete-test.cc:10:0
ncore: 1
core: 
  - /run/mw/delete-test.cc:8:0
    - null pointer dereference
```

**Comment by xiw** [comment]
Use the clang++ you build (rather than the system one) as instructed in INSTALL and you shouldn't see such false positives.

STACK relies on information from that custom clang++ to recognize delete and avoid false positives.

**Comment by wojdyr** [comment]
I'm quite sure I use clang that I've just built. I've built it on Fedora 19 using:

../configure --enable-optimized --disable-assertions --prefix=$HOME/local --enable-cxx11 --enable-targets=host --enable-bindings=none --enable-shared

```
$ clang++ --version
clang version 3.4 (trunk 193953)
Target: x86_64-unknown-linux-gnu
Thread model: posix
```

**Comment by xiw** [comment]
Oops, you'll need `--enable-debug-symbols` to suppress such warnings since STACK needs that from clang to recognize `delete`.  Can you try that?  Thanks.

BTW, just fixed the `INSTALL` file in 81ca1cc77a0647e5308ee7deec4fdb538c4e9086.

**Comment by wojdyr** [comment]
Thanks! I haven't realized that the debug symbols are needed.

I'll test it tonight, but I'm sure it will fix my problem, so I'm closing this issue.
